File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        HashSet<String> set = new HashSet<>();
        HashSet<String> repeat = new HashSet<>();

        for(int i = 0; i + 9 < s.length(); i++) 
        {
            String sub = s.substring(i, i + 10);

            if(!set.contains(sub))
                set.add(sub);
            else
                repeat.add(sub);
        }

        return new ArrayList(repeat);
    }
}

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        map = Counter()
        n = len(s)
        
        for r in range(9,n):
            # create new substring from 0 to 10, last idx is excluding
            subStr = s[r-9: r+1]
            
            map[subStr] += 1
            
        # return all the strings which satisfy the condition
        return [sub for sub in map if map[sub] > 1]